File: two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); i++) {
            if (seen.count(target - nums[i]) != 0) {
                return {seen[target - nums[i]], i};
            }
            else {
                seen[nums[i]] = i;
            }
        }
        return {0, 0};
    }
};

File: two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        HashMap<Integer, Integer> pos = new HashMap<>();
        for (int i = 0; i < n; i++) {
            if (pos.containsKey(target - nums[i])) {
                return new int[]{pos.get(target - nums[i]), i};
            }
            pos.put(nums[i], i);
        }
        return new int[]{-1, -1};
    }
}